packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -176,6 +339,7 @@ export class Interpreter {
   execute (program: string, mode: ExecutionMode, runtime: Runtime, debugStack?: number): void {
     this.runtime = runtime;
     this.instructions = parser(program, mode, this);
+    if (mode === ExecutionMode.APPLICATION) { this.assertValidTxArray(); }
```

Comment:
we were not doing it before v4. shouldn't we consider that here ? i think this check should only be called after AVM is introduced

Reply:
It was before tealv4 as well. We were not doing it. AVM just introduced dynamic size (total len should be 8), and is not version specific (just like pooled transaction fees). Verified on node as well